examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/DiscoveryExampleFragment.java

```diff
@@ -0,0 +1,390 @@
+                              .findFirst();
+                      if (playerInList.isPresent()) {
+                        Log.d(
+                            TAG,
+                            "Replacing existing CastingPlayer entry "
+                                + playerInList.get().getDeviceId()
+                                + " in castingPlayerList list");
+                        arrayAdapter.remove(playerInList.get());
+                      }
+                      arrayAdapter.add(castingPlayer);
+                    });
+          }
+        };
+
+    Button startDiscoveryButton = getView().findViewById(R.id.startDiscoveryButton);
+    startDiscoveryButton.setOnClickListener(
+        v -> {
+          Log.i(
+              TAG, "onViewCreated() startDiscoveryButton button clicked. Calling startDiscovery()");
+          arrayAdapter.clear();
+          if (!startDiscovery(15)) {
+            Log.e(TAG, "onViewCreated() startDiscovery() call Failed");
+          }
+        });
+
+    Button stopDiscoveryButton = getView().findViewById(R.id.stopDiscoveryButton);
+    stopDiscoveryButton.setOnClickListener(
+        v -> {
+          Log.i(TAG, "onViewCreated() stopDiscoveryButton button clicked. Calling stopDiscovery()");
+          stopDiscovery();
+          Log.i(TAG, "onViewCreated() stopDiscoveryButton button clicked. Canceling future task");
+          scheduledFutureTask.cancel(true);
+        });
+
+    Button clearDiscoveryResultsButton = getView().findViewById(R.id.clearDiscoveryResultsButton);
+    clearDiscoveryResultsButton.setOnClickListener(
+        v -> {
+          Log.i(
+              TAG, "onViewCreated() clearDiscoveryResultsButton button clicked. Clearing results");
+          arrayAdapter.clear();
+        });
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    Log.i(TAG, "onResume() called. Calling startDiscovery()");
+    if (!startDiscovery(15)) {
+      Log.e(TAG, "onResume() Warning: startDiscovery() call Failed");
+    }
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    Log.i(TAG, "onPause() called");
+    stopDiscovery();
+    scheduledFutureTask.cancel(true);
+  }
+
+  /** Interface for notifying the host. */
+  public interface Callback {
+    /** Notifies listener of Connection Button click. */
+    // TODO: In following PRs. Implement CastingPlayer connection
+    void handleConnectionButtonClicked(CastingPlayer castingPlayer);
+  }
+
+  private boolean startDiscovery(int discoveryDuration) {
+    Log.i(TAG, "startDiscovery() called, discoveryDuration: " + discoveryDuration + " seconds");
+    if (discovering) {
+      Log.e(TAG, "startDiscovery() Warning: already discovering, stop before starting");
+      return false;
+    }
+    // Add the implemented CastingPlayerChangeListener to listen to changes in the discovered
+    // CastingPlayers
+    MatterError errAdd =
```

Comment:
It is simpler to just declare one variable \`\`\`err\`\`\` and reuse it in the rest of the function rather that creating an errAdd, errStart, etc. Same comment for the stopDiscovery() function below.